### PR TITLE
Bug 1794 bis

### PR DESCRIPTION
**Pull request by hippo91**
This PR fixes the second issue of @nedbat in #1794 .
It consists in avoiding useless exception inference if the exception is not handled.

### CODE REVIEW

**Comment by nedbat** [comment]
Great, thanks! Looking forward to the backport to 1.8 :)

**Comment by hippo91** [comment]
@PCManticore I got pb in the CI with pypy and some tests that are no related to my modifications. 
Have you got an idea?

**Comment by PCManticore** [comment]
@hippo91 Feel free to ignore them for now. I upgraded the pypy version in travis to a version compatible to Python 3, since we started dropping support for Python 2 in master. We can also disable the pypy job for now if you have some time

**Comment by hippo91** [comment]
Thanks @PCManticore .
I will wait for your review and after i merge this patch and backport it to 1.8!

File: pylint/checkers/utils.py [hunk 1]
```diff
@@ -628,17 +629,29 @@ def get_exception_handlers(node, exception):
...
+    (but not under an ExceptHandler!)
+
+    Args:
+        node (astroid.Raise): the node raising the exception.
+
+    Returns:
+        bool: True if the node is inside a try/except statement, False otherwise.
+    """
+    context = _import_node_context(node)
+    if isinstance(context, astroid.TryExcept):
```
Comment (PCManticore) [inline pylint/checkers/utils.py]: `return isinstance(context, astroid.TryExcept)`

**Pull request merged**